Functions/VCS_Info/Backends/VCS_INFO_get_data_git

```diff
@@ -138,8 +138,8 @@ VCS_INFO_git_handle_patches () {
 
 gitdir=${vcs_comm[gitdir]}
 VCS_INFO_git_getbranch ${gitdir}
-gitbase=$( ${vcs_comm[cmd]} rev-parse --show-toplevel )
-rrn=${gitbase:t}
+gitbase=$( ${vcs_comm[cmd]} rev-parse --show-toplevel 2> /dev/null )
+rrn=${gitbase:=$gitdir:t}
```

Comment:
You lost the `:t` modifier in the non-bare repository case.

Could you avoid assigning to `gitbase` on the line that assigns to $rrn, for readability?

```suggestion
if [[ -z $gitbase ]]; then
    gitbase=$gitdir # bare repository
fi
rrn=${gitbase:t}
```